src/components/encoding-pane/property-editor-schema.ts

```diff
@@ -81,34 +109,85 @@ const POSITION_FIELD_TEMPORAL_INDEX = {
   'Axis': ['orient', 'title'].map(p => ({prop: 'axis', nestedProp: p}))
 };
 
-// Capitalize first letter for aesthetic purposes in form
-function toTitleCase(str: string) {
-  return str.replace(/\w\S*/g, txt => {
-    return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase();
-  });
-}
+const COLOR_CHANNEL_FIELD_INDEX = {
+  'Common': [
+    {
+      prop: 'legend',
+      nestedProp: 'title'
+    },
+    {
+      prop: 'scale',
+      nestedProp: 'scheme'
+    }
+  ],
+  'Legend': ['orient', 'title', 'type'].map(p => ({prop: 'legend', nestedProp: p})),
+  'Scale': ['domain', 'scheme', 'type'].map(p => ({prop: 'scale', nestedProp: p}))
+};
 
-function generateTitle(prop: string, nestedProp: string, propTab: string): string {
-  let title;
-  if (propTab === 'Common') {
-    title = nestedProp ? prop + ' ' + nestedProp : prop;
-  } else {
-    title = nestedProp || prop;
-  }
+const SIZE_CHANNEL_FIELD_INDEX = {
+  'Common': [
+    {
+      prop: 'legend',
+      nestedProp: 'title'
+    },
+    {
+      prop: 'scale',
+      nestedProp: 'type'
+    }
+  ],
+  'Legend': ['orient', 'title'].map(p => ({prop: 'legend', nestedProp: p})),
+  'Scale': ['domain', 'range', 'type'].map(p => ({prop: 'scale', nestedProp: p}))
+};
 
-  return toTitleCase(title);
-}
+const SHAPE_CHANNEL_FIELD_INDEX = {
```

Comment:
Not sure if a Common tab is needed here, but if it does it's easy to add in.

Reply:
If you gonna have two tabs with two properties each, better just have one tab with 4 properties.  Same for other channels :)

Reply:
Sounds good. This means each positional channel will only have one common tab, since there are only 4 properties (for right now, will add more later).

![image](https://user-images.githubusercontent.com/9298611/39677776-92e65994-5135-11e8-9e47-1efde1ab80e4.png)


Is this ok?

Reply:
I guess it is ok.  The current state is a bit funny since 1 tab = better have no tab too.

But we will add more pretty soon, so let's just keep the tab. :)